src/bin/sol-fbp-runner/main.c

```diff
@@ -149,12 +181,194 @@ parse_args(int argc, char *argv[])
     return true;
 }
 
+#ifdef USE_MEMMAP
+static void
+clear_memory_maps(void)
+{
+    const struct sol_str_table_ptr *iter;
+    struct sol_memmap_map *map;
+    int i;
+
+    SOL_PTR_VECTOR_FOREACH_IDX (&memory_maps, map, i) {
+        for (iter = map->entries; iter->key; iter++) {
+            free((void *)iter->val);
+        }
+
+        free(map);
+    }
+
+    sol_ptr_vector_clear(&memory_maps);
+}
+
+static bool
+load_memory_map_file(const char *json_file)
+{
+    struct sol_file_reader *fr = NULL;
+    struct sol_json_scanner scanner, entries;
+    struct sol_str_slice contents;
+    struct sol_json_token token, key, value;
+    enum sol_json_loop_reason reason;
+    struct sol_vector entries_vector = SOL_VECTOR_INIT(struct sol_str_table_ptr);
+    struct sol_memmap_map *map;
+    struct sol_memmap_entry *memmap_entry;
+    struct sol_str_table_ptr *ptr_table_entry;
+    uint32_t version;
+    size_t entries_vector_size;
+    void *data;
+    int i;
+    char *path;
+
+    fr = sol_file_reader_open(json_file);
+    if (!fr) {
+        SOL_ERR("Couldn't open json file '%s': %s\n", json_file, sol_util_strerrora(errno));
+        return false;
+    }
+
+    contents = sol_file_reader_get_all(fr);
+    sol_json_scanner_init(&scanner, contents.data, contents.len);
+
+    SOL_JSON_SCANNER_ARRAY_LOOP (&scanner, &token, SOL_JSON_TYPE_OBJECT_START, reason) {
+        SOL_JSON_SCANNER_OBJECT_LOOP_NEST (&scanner, &token, &key, &value, reason) {
+            if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "path")) {
+                sol_json_token_remove_quotes(&value);
+                path = strndupa(value.start,
+                    sol_json_token_get_size(&value));
+                if (!path) {
+                    SOL_WRN("Couldn't get map path");
+                    goto error;
+                }
+            } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "version")) {
+                if (sol_json_token_get_uint32(&value, &version) < 0) {
+                    SOL_WRN("Couldn't get memory map version");
+                    goto error;
+                }
+            } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "entries")) {
+                sol_json_scanner_init_from_token(&entries, &value);
+
+                SOL_JSON_SCANNER_ARRAY_LOOP (&entries, &token, SOL_JSON_TYPE_OBJECT_START, reason) {
+                    uint32_t size = 0, offset = 0, bit_offset = 0, bit_size = 0;
+                    char *name = NULL;
+
+                    SOL_JSON_SCANNER_OBJECT_LOOP_NEST (&entries, &token, &key, &value, reason) {
+                        if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "name")) {
+                            sol_json_token_remove_quotes(&value);
+                            name = strndupa(value.start,
```

Comment:
I don't think it'll be a problem here, but please avoid calling `alloca()` inside a loop. It's good practice, as the allocated memory is tied to the call frame and the stack might overflow. For this case, since the JSON data is still in RAM, you might want to use a string slice and use `sol_arena_strdup_slice()`.